File: itx-go/itx-worker/internal/compute/dispatcher.go
package compute

import (
	"context"
	"log/slog"
)

type Dispatcher struct {
	state WorkerState
}

func NewDispatcher(state WorkerState) *Dispatcher {
	return &Dispatcher{state: state}
}

func (d *Dispatcher) Handle(_ context.Context, body string) error {
	// No compute-plane message types yet — log and ack so the queue stays drained.
	slog.Info("compute message received (no handler yet)", "body", body)
	return nil
}

File: itx-rs/itx-worker/src/compute/dispatcher.rs
use async_trait::async_trait;
use itx_contract::queue::{HandlerError, MessageHandler};

use crate::compute::state::ComputeWorkerState;

pub struct ComputeDispatcher {
    #[allow(dead_code)] // wired up by upcoming handlers (e.g. send-notification → email API)
    state: ComputeWorkerState,
}

impl ComputeDispatcher {
    pub fn new(state: ComputeWorkerState) -> Self {
        Self { state }
    }
}

#[async_trait]
impl MessageHandler for ComputeDispatcher {
    async fn handle(&self, body: &str) -> Result<(), HandlerError> {
        // No compute-plane message types yet — just log and ack so the queue stays drained.
        tracing::info!(body = body, "compute message received (no handler yet)");
        Ok(())
    }
}
